src/api/store/vendor.py

```diff
@@ -196,13 +223,23 @@ def copy_vendor(self, vendor_id) -> Tuple[Vendor, MassEnergizeAPIError]:
       vendor: Vendor = Vendor.objects.get(id=vendor_id)
       if not vendor:
         return CustomMassenergizeError(f"No vendor with id {vendor_id}")
-        
-      vendor.pk = None
-      vendor.is_published = False
-      vendor.is_verified = False
-      vendor.name = f"{vendor.name}-Copy-{random.randint(1,100000)}"
-      vendor.save()
-      return vendor, None
+
+      old_tags = vendor.tags.all()
+      new_vendor = vendor
+
+      new_vendor.pk = None
+      new_vendor.is_published = False
+      new_vendor.is_verified = False
+
+      new_name = _copy_title(vendor.name)
```

Comment:
i see that you got rid of the random number generator.  I'm not sure the replacement is the best path forward here either.